kobo/hub/views.py

```diff
@@ -274,3 +274,15 @@ def krb5login(request, redirect_field_name=REDIRECT_FIELD_NAME):
         redirect_to = reverse("home/index")
     return RedirectView.as_view(url=redirect_to, permanent=True)(request)
 
+def oidclogin(request):
+    middleware = 'kobo.django.auth.middleware.LimitedRemoteUserMiddleware'
+    if django_version_ge('1.10.0'):
+        middleware_setting = settings.MIDDLEWARE
+    else:
+        middleware_setting = settings.MIDDLEWARE_CLASSES
+    if middleware not in middleware_setting:
+        raise ImproperlyConfigured("oidclogin view requires '%s' middleware installed" % middleware)
+    return RedirectView.as_view(url=reverse("home/index"), permanent=False)(request)
+
+def oidc_redirect(request):
+    return HttpResponse("Redirecting...")
```

Comment:
The implementation of this doesn't make sense to me.

- How is it redirecting?  The code doesn't specify any status code associated with a redirect (3xx).
- Where is it redirecting to?  It's not specifying any Location header to be set in the response.

Are you sure this code is ever reached?  From what I've read in the mod_auth_openidc docs, I'm under the impression that the request for this URL will be processed entirely within httpd and the request won't make it to the WSGI layer.  I thought that's why the docs say:

```
# OIDCRedirectURI is a vanity URL that must point to a path protected by this module but must NOT point to any content
```

i.e. "must NOT point to any content", because processing of the request is going to be stopped within mod_auth_openidc so there will not be any chance to serve content.

If this code is actually reached, can you explain a bit how it works?

Reply:
You're right, I think I made an incorrect assumption about how the authentication works. The URL is not accessed by the user's browser, and no redirect logic is necessary to route the user back to the original URL. mod_auth_openidc's docs call it a "redirect url", so I guess I just went with that. I changed the response to not be misleading. I think it's okay to keep the URL name as "redirect", since this is how mod_auth_openidc calls it. What do you think?

Reply:
I asked "Are you sure this code is ever reached" but from your response I'm still not getting whether the code is reached.

Have you tried *not* adding this code, and seeing if it still works?

I currently suspect that the route and view for oidclogin/redirect/ don't achieve anything, and if I'm right then I don't think it should be added as it only misleads people as to how things actually work.  If I'm wrong, and removing this code breaks it, I would appreciate if you can share some details *how* it's broken, that would help me understand why I'm wrong.

Reply:
Ah, sorry I misunderstood. I tested it and the vanity URL can be set to any path protected by mod_auth_openidc. I removed the extra URL as it's not necessary and modified the commit message.